### ISSUE TITLE
docs_publish is failing due to downloading AARCH64 binary of firebase tool on x64 bot

### ISSUE THREAD

**Issue by christopherfujino** [flutter/flutter#134319]
On https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20docs_publish/11997/overview

Which ran on an x64 ubuntu bot: https://chromium-swarm.appspot.com/bot?id=flutter-prod-ubuntu-highmem-us-central1-b-4-0kd9

The script failed with:

```
/lib/ld-linux-aarch64.so.1: No such file or directory
```

https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8770491697891583569/+/u/Docs/Deploy_docs/Firebase_deploy/stdout

After it downloaded an AArch64 binary from https://firebase.tools/bin/linux/latest: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8770491697891583569/+/u/Install_firebase/Install_firebase_bin/execution_details

Our recipes have CURL'd this endpoint since 2020: https://flutter.googlesource.com/recipes/+/59aeb93b9bc8cf5cf6ce9a499cc9703cb734f58f. There have been no recent changes. I'm guessing that the issue is that https://firebase.tools/bin/linux/latest used to serve an x64 binary, but now switched to ARM.

**Comment by christopherfujino** [comment]
Unblocking the tree while this is investigated: https://github.com/flutter/flutter/pull/134320. cc @itsjustkevin we'll want to make sure this is fixed before our next release.

**Comment by shihaohong** [comment]
I tried running `objdump` on the binary from https://firebase.tools/bin/linux/latest to see, it seems to be generating an x86-64 ELF binary.

some details about the binary i downloaded:
```
Size: 150,097,398 bytes (150.1 MB on disk)
Created: Friday, September 8, 2023 at 2:21 PM
```

```sh
shihao$ objdump -p "firebase-tools-linux" | head

firebase-tools-linux:	file format elf64-x86-64

Program Header:
    PHDR off    0x0000000000000040 vaddr 0x0000000000400040 paddr 0x0000000000400040 align 2**3
         filesz 0x00000000000002d8 memsz 0x00000000000002d8 flags r--
  INTERP off    0x0000000000000318 vaddr 0x0000000000400318 paddr 0x0000000000400318 align 2**0
         filesz 0x000000000000001c memsz 0x000000000000001c flags r--
    LOAD off    0x0000000000000000 vaddr 0x0000000000400000 paddr 0x0000000000400000 align 2**12
         filesz 0x00000000004c0810 memsz 0x00000000004c0810 flags r--
```

**Comment by shihaohong** [comment]
Also linking the latest version (as of writing this) of firebase-tools for convenience: https://github.com/firebase/firebase-tools/releases/tag/v12.5.3

**Comment by christopherfujino** [comment]
When I first looked at this, I had an arm64 binary (which i unfortunately deleted), but now when I curl the same endpoint I get an x64 binary (the same as @shihaohong), so maybe this was fixed upstream?

**Comment by christopherfujino** [comment]
Re-ran the build ONE MORE TIME for science: https://ci.chromium.org/p/flutter/builders/prod/Linux%20docs_publish/11999?

**Comment by christopherfujino** [comment]
That build passes, so it appears the upstream issue was resolved. I will leave this issue open to track pinning this dependency however, to avoid future outages.

**Comment by yusuf-goog** [comment]
lowering priority on this since its not an emergent issue.

**Comment by yusuf-goog** [comment]
@christopherfujino should this be removed from the infra queue?

**Comment by christopherfujino** [comment]
> @christopherfujino should this be removed from the infra queue?

Yeah, this sounds fair as it is not currently blocking the tree. I'm not currently working on it either, but I think we should track this as tech debt because it could happen again.
